engine/source_sym_engine.ml

```diff
@@ -213,7 +213,7 @@ let sym_exec source =
   let type_env = Source_env.build_type_env source.type_decls in
   let rec decompose (matrix : matrix) : constraint_tree =
     match matrix with
-    | (_, []) -> assert false
+    | (_, []) -> Failure
```

Comment:
Which this definition you can remove the emptiness check on `fallback` on lines 238-239.

Reply:
Good catch, I overlooked